File: 0669-trim-a-binary-search-tree/0669-trim-a-binary-search-tree.py
class Solution:
    def trimBST(self, root: Optional[TreeNode], low: int, high: int) -> Optional[TreeNode]:
        if root is None:
            return None
        if root.val<low:
            return self.trimBST(root.right,low,high)
        elif high<root.val:
            return self.trimBST(root.left,low,high)
        root.left = self.trimBST(root.left,low,high)
        root.right = self.trimBST(root.right,low,high)
        return root

File: 0669-trim-a-binary-search-tree/0669-trim-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode trimBST(TreeNode root, int low, int high) {
        if(root==null)
        {
            return null;
        }
        else if((int)root.val<low)
        {
            // root.left=null;
            
            return trimBST(root.right,low,high);
        }
        else if((int)high<root.val)
        {
            // root.right=null;
            
            return trimBST(root.left,low,high);
        }
        else
        {
            root.left=trimBST(root.left,low,high);
            root.right=trimBST(root.right,low,high);
            return root;

        }
    }
}